common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/POIEntities.java

```diff
@@ -38,100 +36,104 @@ public class POIEntities {
         if (!interval.isReady()) return;
 
-        try {
-            MinecraftClient minecraftClient = MinecraftClient.getInstance();
-
-            if (minecraftClient == null) return;
-            if (minecraftClient.player == null) return;
-            if (minecraftClient.world == null) return;
-            if (minecraftClient.currentScreen != null) return; //Prevent running if any screen is opened
+        MinecraftClient minecraftClient = MinecraftClient.getInstance();
 
-            for (POIGroup group : builtInGroups.values()) {
-                group.clearEntities();
-            }    
+        if (minecraftClient == null) return;
+        if (minecraftClient.player == null) return;
+        if (minecraftClient.world == null) return;
+        if (minecraftClient.currentScreen != null) return; //Prevent running if any screen is opened
 
-            log.debug("POIEntities started.");
+        log.debug("POIEntities started.");
 
-            // Copied from PlayerEntity.tickMovement()
-            Box scanBox = minecraftClient.player.getBoundingBox().expand(range, range, range);
-            List<Entity> entities = minecraftClient.world.getOtherEntities(minecraftClient.player, scanBox);
+        for (POIGroup<Entity> group : groups) {
+            group.clear();
+        }
 
-            if (onPOIMarkingNow && POIMarkingConfigMap.getInstance().isSuppressOtherWhenEnabled()) {
-                POIGroup passiveGroup = builtInGroups.get("passive");
-                POIGroup hostileGroup = builtInGroups.get("hostile");
+        Box scanBox = minecraftClient.player.getBoundingBox().expand(range, range, range);
+        List<Entity> entities = minecraftClient.world.getOtherEntities(minecraftClient.player, scanBox);
 
-                for (Entity e : entities) {
-                    if (this.markedEntity.test(e)) {
-                        if (passiveGroup.checkAndAddEntity(markedEntity))
-                            this.playSoundAt(e.getBlockPos(), passiveGroup.sound, passiveGroup.soundPitch);
-                        if (hostileGroup.checkAndAddEntity(markedEntity))
-                            this.playSoundAt(e.getBlockPos(), hostileGroup.sound, hostileGroup.soundPitch);
-                    }
-                }
-
-                return;
-            }
+        for (POIGroup<Entity> group : groups) {
+            entities.removeIf(group::add);
+        }
 
-            for (POIGroup group : builtInGroups.values()) {
-                for (Entity e : entities) {
-                    if (group.checkAndAddEntity(e)) {
-                        this.playSoundAt(e.getBlockPos(), group.sound, group.soundPitch);
-                        entities.remove(e);
-                    }
+        for (POIGroup<Entity> group : groups) {
+            for (Entity entity : group.getItems()) {
+                if (isMarking && POIMarkingConfigMap.getInstance().isSuppressOtherWhenEnabled() && !(marked == null || marked.isInstance(entity))) {
```

Comment:
Sorry but I can't understand the meaning of the `!(marked == null || marked.isInstance(entity))`, could you clarify it with a boolean variable?

Reply:
That checks if the entity is not an instance of the marked entity and thus should be silenced. It also has a simple null check thrown in because marked is nullable.

Reply:
if `POI marking on` and `suppress enabled` and, `marked != null` and `the entity is NOT the marked type`, yeah, I understand now, does IntelliJ automatically suggested extracting the `not`? I remember I learned the knowledge in university...

Reply:
Intellij did not have any suggestions when I wrote it. I'm also not quite sure what you mean.